AlteracPullAnnouncer.lua

```diff
@@ -42,54 +73,44 @@ function frame:OnEvent(event, ...)
         return
     end
 
-    local subevent, _, _, sourceName, _, _, _, destName = select(2, ...)
-    local spellId, spellName
+    if not subEvents[subevent] then return end
 
-    if destName == sourceName then
-        return
-    end
+    -- assignment is cheap, function calls not, select() included, we call CombatLogGetCurrentEventInfo after we pass the trivial tests
+    local timestamp, subevent, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, arg12, arg13, arg14, arg15, arg16, arg17, arg18, arg19, arg20, arg21, arg22, arg23, arg24 = CombatLogGetCurrentEventInfo()
 
-    if isBossOrGuard(destName) then
-        if isPulled == false then
-            if stringstarts(subevent, "SPELL") then
-                spellName = select(13, ...)
+    local dstUnitType,_,_,_,_, dstID = strsplit("-", destGUID)
+    local srcUnitType,_,_,_,_, srcID = strsplit("-", sourceGUID)
 
-                pullAnnounce(destName, sourceName, PULL_ACTION, spellName)
-            elseif subevent == "SWING_DAMAGE" or subevent == "SWING_MISSED" then
-                pullAnnounce(destName, sourceName, PULL_ACTION, MELEE)
+    if dstUnitType == "Creature" and NPCids[dstID] then -- boss or guard getting attacked
+        if not isPulled then
+            if strfind(subevent, "^SPELL") or strfind(subevent, "^RANGE") then -- ^ anchors the search term at the start
+                local spellId, spellName = arg12, arg13 -- use spellName so we don't have to list all rank spellids
+                if not nonCombatSpells[spellName] then
+                    addon:pullAnnounce(destName, sourceName, PULL_ACTION, spellName)
+                end
+            elseif strfind(subevent, "^SWING") then -- melee
+                addon:pullAnnounce(destName, sourceName, PULL_ACTION, MELEE)
             end
-        else
-            start = GetTime()
-        end
-    elseif isPulled == true then
-        if start ~= nil and GetTime() - start > 20 then
-            -- debug
-            DEFAULT_CHAT_FRAME:AddMessage("20 sec elapsed, setting pull to false")
-            isPulled = false

```

Comment:
This is guess is missing now. It never resets the pull? So assuming you are pulling, but then running out and the fight resets. we are not resetting the addon in that case.